### DISCUSSION TITLE
Trying to use online postgresql service for the database url

### DISCUSSION THREAD

**Discussion by mooyg** [prisma/prisma#7413] [Q&A] [answered]
When trying to use some online postgresql such as heroku postgres/elephantsql I am always getting 
```
Prisma Migrate could not create the shadow database. Please make sure the database user has permission to create databases
```
https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database here they suggest putting this in as shadow database URL but when using things like online IDE it won't have a database on local. Does someone have any idea what to do in this case -> When postgresql isn't installed on local machine or using some online IDE :D. 
```
Prisma Migrate could not create the shadow database. Please make sure the database user has permission to create databases. More info: https://pris.ly/d/migrate-shadow. Original error:
Database error
Error querying the database: db error: ERROR: permission denied to create database
   0: sql_migration_connector::flavour::postgres::sql_schema_from_migration_history
             at migration-engine\connectors\sql-migration-connector\src\flavour\postgres.rs:274
   1: sql_migration_connector::sql_database_migration_inferrer::calculate_drift
             at migration-engine\connectors\sql-migration-connector\src\sql_database_migration_inferrer.rs:57
   2: migration_core::api::DevDiagnostic
             at migration-engine\core\src\api.rs:95
``` 
(When I was using typeorm I never faced this issue)

**Comment by ryands17** [comment, marked as answer]
@mooyg 👋 

As suggested in the doc, you need to create another database and pass the database URL in the `shadowDatabaseUrl` parameter as shown here: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database#cloud-hosted-shadow-databases-must-be-created-manually

This is the only way to perform this for cloud hosted databases.

**Reply by mooyg** [reply]
And what If the machine I am using doesn't have postgres and completely depends on the cloud provider?

**Reply by ryands17** [reply]
You can create a new database on the cloud provider and use that as the shadow database. If you still cannot create a shadow database on the cloud provider, then you can use version _2.12.0_ of `prisma` and `@prisma/cli` for the time being.

**Reply by mooyg** [reply]
Ohk thanks for your response 👍🏻

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
